File: frontend/src/lib/aiModels/videoModels.ts
import type { AIModel } from '../aiModels';

export const VIDEO_MODELS: AIModel[] = [
  { id: 'veo-3-generate', name: 'Veo 3 Generate', category: 'video', capabilities: ['video-gen'], description: 'Veo video generation core', isFree: false },
  { id: 'veo-3-fast-generate', name: 'Veo 3 Fast Generate', category: 'video', capabilities: ['video-gen'], description: 'High-speed Veo video generator', isFree: true },
  { id: 'veo-3-lite-generate', name: 'Veo 3 Lite Generate', category: 'video', capabilities: ['video-gen'], description: 'Lite Veo compilation node', isFree: true },
  { id: 'stable-video-diffusion', name: 'Stable Video Diffusion', category: 'video', capabilities: ['video-gen'], description: 'Open video diffusion baseline', isFree: true },
  { id: 'cogvideox', name: 'CogVideoX', category: 'video', capabilities: ['video-gen'], description: 'Open video generation model', isFree: true },
  { id: 'open-sora-plan', name: 'Open-Sora-Plan', category: 'video', capabilities: ['video-gen'], description: 'Open video generation planning model', isFree: true },
  { id: 'easyanimate', name: 'EasyAnimate', category: 'video', capabilities: ['video-gen'], description: 'Animated video generation pipeline', isFree: true },
  { id: 'animatediff', name: 'AnimateDiff', category: 'video', capabilities: ['video-gen'], description: 'Animation diffusion generator', isFree: true },
];

export default VIDEO_MODELS;

File: backend/lib/video_models.py
"""
Anime Script Pro — Video Generation Model Registries

This module lists supported model paths for dynamic scene animating and video compilation.

Sections (in order):
  1. Registry Definitions
  2. Fallback Assignments
"""

# ==============================================================================
# 1. REGISTRY DEFINITIONS
# ==============================================================================
VIDEO_MODEL_IDS = [
    "veo-3.1-fast-generate-preview",
    "veo-3-generate",
    "veo-3-fast-generate",
    "veo-3-lite-generate",
    "stable-video-diffusion",
    "cogvideox",
    "open-sora-plan",
    "easyanimate",
    "animatediff",
]

FREE_VIDEO_MODEL_IDS = [
    "veo-3.1-fast-generate-preview",
    "veo-3-fast-generate",
    "veo-3-lite-generate",
    "stable-video-diffusion",
    "cogvideox",
    "open-sora-plan",
    "easyanimate",
    "animatediff",
]

PAID_VIDEO_MODEL_IDS = [
    "veo-3-generate",
]

# ==============================================================================
# 2. FALLBACK ASSIGNMENTS
# ==============================================================================
DEFAULT_VIDEO_MODELS = FREE_VIDEO_MODEL_IDS

__all__ = [
    "VIDEO_MODEL_IDS",
    "FREE_VIDEO_MODEL_IDS",
    "PAID_VIDEO_MODEL_IDS",
    "DEFAULT_VIDEO_MODELS",
]
